Inventory edit - merge

MariskaIJpelaar opened the pull request.

MariskaIJpelaar committed:
Finished editing feature of the inventory tab

MariskaIJpelaar committed:
Changed the layout of the edit activity of the inventory tab.

MariskaIJpelaar committed:
Changed layout of all inventory tabs

Sebastiaan-Alvarez-Rodriguez commented:
Functionality is in order, but the layouts must be altered to be more user-friendly.
Mainly, the [activity_inventory_add.xml](https://github.com/Sebastiaan-Alvarez-Rodriguez/Xenopelthis/blob/inventory_edit_branch/app/src/main/res/layout/activity_inventory_add.xml) must change:
A Spinner is not a good idea, when the displayed list may grow very long. Instead, use a Recyclerview and show all products (except those who are already added to inventory of course). After this, you could forward chosen product to the *editActivity, where the user may choose an amount.

Good work thus far!

Sebastiaan-Alvarez-Rodriguez reviewed (changes requested):
Overall code looks good. However, please change the use of spinners to full RecyclerViews, and make adding an item to the inventory a 2-step process (click item, set amount). This way, you could recycle the 'set amount' step for edit purposes + you get a handy recyclerview instead of a not-so-handy spinner.

Sebastiaan-Alvarez-Rodriguez committed:
Merge pull request #27 from Sebastiaan-Alvarez-Rodriguez/master

Enhance-merge

Sebastiaan-Alvarez-Rodriguez commented:
Please fix problems created by template adaption in #26 before merge

MariskaIJpelaar committed:
Changed layout of InventoryAddActivity.
 - First an inventory_item is selected to add with amount zero
 - Then the user can set the amount through the InventoryEditActivity

 Also added a Searcher for the inventory (both Fragment and Add activity)

Sebastiaan-Alvarez-Rodriguez committed:
Changes in inventory UI and controllers to be more user friendly

Sebastiaan-Alvarez-Rodriguez reviewed (approved):
I updated some behaviors to remove deprecated code, and allow standard patterns to be reused. It is more user-friendly now
